JS/Lesson34/goksel/recipe-app/src/modules/user/LogIn/index.tsx

```diff
@@ -0,0 +1,132 @@
+
+                <Box component="form" noValidate sx={{ mt: 3 }} onSubmit={handleSubmit(onSubmit)}>
+                    <Grid container spacing={2}>
+
+                        <Grid item xs={12}>
+                            <TextField
+                                id='email'
+                                label="Email"
+                                variant='outlined'
+                                size='medium'
+                                type='email'
+                                required
+                                fullWidth
+                                helperText="Please Enter Your Email"
+                                {...register("email")}
+                            />
+
+                        </Grid>
+
+                        <Grid item xs={12}>
+                            <FormControl
+                                variant='outlined'
+                                required
+                                size='medium'
+                                fullWidth
+                                {...register("password")}
+                            >
+                                <InputLabel htmlFor="password">Password</InputLabel>
+                                <OutlinedInput
+                                    id="password"
+                                    type={showPassword ? "text" : "password"}
+                                    endAdornment={
+                                        <InputAdornment position='end'>
+                                            <IconButton
+                                                aria-label="toggle password visibility"
+                                                onClick={handleClickShowPassword}
+                                                edge="end"
+                                            >
+                                                {showPassword ? <VisibilityOff /> : <Visibility />}
+                                            </IconButton>
+                                        </InputAdornment>
+                                    }
+                                />
+                                <FormHelperText>Please Enter Your Password</FormHelperText>
+                            </FormControl>
+                        </Grid>
+
+                        <Grid item>
+                            <FormControlLabel
+                                control={<Checkbox value="remember" color='primary' />}
+                                label="Remember me"
+                            />
+                        </Grid>
+
+                        <Grid item xs={12}>
+                            <Link to="/">
+                                <Button
+                                    type='submit'
+                                    fullWidth
+                                    variant='contained'
+                                    onClick={onSubmit}
```

Comment:
Remove `onClick={onSubmit}`from this button. The form element itself already has an onSubmit event handler.

Reply:
I removed it but submit function is not working now.

Reply:
When ı remove it onSubmit function was not working and ı removed Link element as well and ı used useNavigate from react-router-dom for to navigate the page now it's working well.